common/util/sha256.h

```diff
@@ -0,0 +1,138 @@
+/***************** See RFC 6234 for details. *******************/
+/*
+   Copyright (c) 2011 IETF Trust and the persons identified as
+   authors of the code.  All rights reserved.
+
+   Redistribution and use in source and binary forms, with or
+   without modification, are permitted provided that the following
+   conditions are met:
+
+   - Redistributions of source code must retain the above
+     copyright notice, this list of conditions and
+     the following disclaimer.
+
+   - Redistributions in binary form must reproduce the above
+     copyright notice, this list of conditions and the following
+     disclaimer in the documentation and/or other materials provided
+     with the distribution.
+
+   - Neither the name of Internet Society, IETF or IETF Trust, nor
+     the names of specific contributors, may be used to endorse or
+     promote products derived from this software without specific
+     prior written permission.
+
+   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+   CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+   INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+   MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+   DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+   CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+   SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+   NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+   LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+   HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+   CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+   OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+   EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+   Implementation of the SHA256 Secure Hash Algorithm as defined in the U.S.
+   National Institute of Standards and Technology Federal Information Processing
+   Standards Publication (FIPS PUB) 180-3 published in October 2008 and formerly
+   defined in its predecessors, FIPS PUB 180-1 and FIP PUB 180-2. See:
+
+   The algorithm description is available at
+   http://csrc.nist.gov/publications/fips/fips180-3/fips180-3_final.pdf
+
+   The code is derived from: https://www.rfc-editor.org/rfc/rfc6234.txt
+*/
+
+#ifndef VERIBLE_COMMON_UTIL_SHA_H_
+#define VERIBLE_COMMON_UTIL_SHA_H_
+
+#include <array>
+#include <cstdint>
+#include <string>
+
+#include "absl/strings/string_view.h"
+
+namespace verible {
+
+// The size of SHA256 hash in bytes.
+constexpr int32_t kSha256HashSize = 32;
```

Comment:
Using `inline constexpr` in the header makes sure that there is only one copy of this constant in the binary.

Reply:
Done